Tests/phpunit/ConfigValidatorTest.php

```diff
@@ -0,0 +1,404 @@
+        $validator = new ConfigValidator();
+        $validator->validate($logger);
+
+        self::assertEquals('', $validator->getScript());
+        self::assertEquals('/data/', $validator->getDataDir());
+        self::assertEquals(['tables' => [], 'files' => []], $validator->getInput());
+        self::assertContains('987654321', $validator->getRunId());
+        self::assertNotEmpty($validator->getClient());
+    }
+
+    public function testNoConfig(): void
+    {
+        $logger = new Logger('test', [new NullHandler()]);
+        $validator = new ConfigValidator();
+        self::expectException(InvalidInputException::class);
+        self::expectExceptionCode(170);
+        self::expectExceptionMessage('Environment KBC_EXPORT_CONFIG is empty.');
+        $validator->validate($logger);
+    }
+
+    public function testInvalidPlainConfigJson(): void
+    {
+        putenv('KBC_EXPORT_CONFIG={"a"b}');
+        $logger = new Logger('test', [new NullHandler()]);
+        $validator = new ConfigValidator();
+        self::expectException(InvalidInputException::class);
+        self::expectExceptionCode(0);
+        self::expectExceptionMessage('Input configuration is invalid: Syntax error');
+        $validator->validate($logger);
+    }
+
+    public function testInvalidPlainConfig(): void
+    {
+        $configuration = ['bad-node' => ['input' => []]];
+        putenv('KBC_EXPORT_CONFIG=' . json_encode($configuration));
+        $logger = new Logger('test', [new NullHandler()]);
+        $validator = new ConfigValidator();
+        self::expectException(InvalidInputException::class);
+        self::expectExceptionCode(171);
+        self::expectExceptionMessage(
+            'Configuration is invalid: Unrecognized option "bad_node" ' .
+            'under "configuration". Available option is "storage".'
+        );
+        $validator->validate($logger);
+    }
+
+    public function testInvalidTransformationConfig1(): void
+    {
+        putenv('KBC_CONFIG_ID=123');
+        $logger = new Logger('test', [new NullHandler()]);
+        $validator = new ConfigValidator();
+        self::expectException(InvalidInputException::class);
+        self::expectExceptionCode(170);
+        self::expectExceptionMessage('Environment KBC_CONFIG_ID or KBC_ROW_ID or KBC_CONFIG_VERSION is empty.');
+        $validator->validate($logger);
+    }
+
+    public function testInvalidTransformationConfig2(): void
+    {
+        putenv('KBC_CONFIG_ID=123');
+        putenv('KBC_ROW_ID=123');
+        $logger = new Logger('test', [new NullHandler()]);
+        $validator = new ConfigValidator();
+        self::expectException(InvalidInputException::class);
+        self::expectExceptionCode(170);
+        self::expectExceptionMessage('Environment KBC_CONFIG_ID or KBC_ROW_ID or KBC_CONFIG_VERSION is empty.');
+        $validator->validate($logger);
+    }
+
+    public function testInvalidTransformationConfig3(): void
```

Comment:
Sorry, this is nit-picking, but can we call these methods by what they are testing?  Example `testInvalidTransformationConfigMissingRowId` etc?